source/Engine.cpp

```diff
@@ -2441,12 +2441,20 @@ void Engine::DoCollection(Flotsam &flotsam)
 	}
 
 	// Unless something went wrong while forming the message, display it.
-	if(!message.empty())
-	{
-		int free = collector->Cargo().Free();
+	if(message.empty()) return;
+
+	int free = collector->Cargo().Free();
+	int total = 0;
+	for(const shared_ptr <Ship> &ship : player.Ships())
+		if(!ship->IsParked() && ship->GetSystem() == player.GetSystem())
+			total += ship->Cargo().Free();
+
+	if(free == total)
 		message += " (" + Format::CargoString(free, "free space") + " remaining.)";
```

Comment:
Maybe move this line (and a similar one from the `else` block) outside the `if..else` to avoid duplicating code?

Reply:
Both options are fine for me, but probably some people would say the second one is clearer.